crates/shell/src/shells/elvish.rs

```diff
@@ -24,11 +24,11 @@ fn format(value: impl AsRef<str>) -> String {
 // https://elv.sh/ref/command.html#using-elvish-interactivelyn
 impl Shell for Elvish {
     fn format_env_set(&self, key: &str, value: &str) -> String {
-        format!("set-env {key} {}", format(value))
+        format!("set-env {} {}", self.quote(key), format(value))
```

Comment:
2 things:
- Add a semi colon (looks like I forgot it)
- We probably need to integrate quoting with the `format` function